model_analyzer/result/sorted_results.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional
+
+from copy import deepcopy
+from bisect import insort
+
+from model_analyzer.constants import LOGGER_NAME
+from model_analyzer.result.run_config_result import RunConfigResult
+
+import logging
+
+logger = logging.getLogger(LOGGER_NAME)
+
+
+class SortedResults:
+    """
+    A data structure used by the result manager 
+    to store and sort results
+    """
+
+    GET_ALL_RESULTS = -1
+
+    def __init__(self):
+        self._run_config_results = []
+
+    def results(self) -> List[RunConfigResult]:
+        """
+        Returns
+        -------
+        All the results
+        """
+
+        self._run_config_results.sort()
+        return self._run_config_results
+
+    def add_result(self, run_config_result: RunConfigResult):
+        """
+        Adds a run_config_result to the result lists
+        This can either be a new result or new measurements added
+        to an existing result
+
+        Parameters
+        ----------
+        result: ModelResult
+            The result to be added
+        """
+
+        existing_run_config_result = self._find_existing_run_config_result(
+            run_config_result)
+
+        if existing_run_config_result:
+            self._add_measurements_to_existing_run_config_result(
+                existing_run_config_result, run_config_result)
+        else:
+            self._add_new_run_config_result(run_config_result)
+
+    def top_n_results(self, n: int) -> List[RunConfigResult]:
+        """
+        Parameters
+        ----------
+        n : int
+            The number of  top results
+            to retrieve
+
+        Returns
+        -------
+        list of RunConfigResults
+            The n best results for this model,
+            must all be passing results
+        """
+        self._run_config_results.sort()
```

Comment:
move this sort into the function on the next line, so we are guaranteed that anyone who uses that function will get sorted results.

If you need to, for clarity, you can rename the function to indicate it does a sort

Reply:
I don't understand. The other function is private. I believe the sort is in the correct location.

Reply:
I've moved it, but I don't understand why it needs to be in the private method.